src/ILCompiler.Compiler/src/IL/ILImporter.Scanner.cs

```diff
@@ -275,7 +277,7 @@ private void ImportCall(ILOpcode opcode, int token)
 
             // If we're scanning the fallback body because scanning the real body failed, don't trigger cctor.
             // Accessing the cctor could have been a reason why we failed.
-            if (!_isFallbackBodyCompilation)
+            if (!_isFallbackBodyCompilation && !method.Signature.IsStatic)
```

Comment:
Why do we need the `!method.Signature.IsStatic` condition? I would expect it to be opposite - add cctor reference for static methods and regular constructors.